### PR TITLE
arhetype+models(with relations)+travis

### PR DESCRIPTION
**Pull request by aslovyagin**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+language: java
+branches:
+    only:
+      - master
+      - developer
```
Comment (M0rtyMerr) [inline .travis.yml:5]: Why only them?

File: src/main/java/com/epam/controller/IndexController.java [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
 
-import java.time.LocalDateTime;
-import java.util.HashMap;
-import java.util.Map;
 
 @Controller
+@RequestMapping(path="/demo")
```
Comment (M0rtyMerr) [inline src/main/java/com/epam/controller/IndexController.java:15]: Without "path=" everywhere

File: src/main/java/com/epam/controller/IndexController.java [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
-import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
 
-import java.time.LocalDateTime;
-import java.util.HashMap;
-import java.util.Map;
 
 @Controller
```
Comment (M0rtyMerr) [inline src/main/java/com/epam/controller/IndexController.java:14]: -> `@RestController`

File: src/main/java/com/epam/controller/IndexController.java [hunk 4]
```diff
@@ -1,19 +1,34 @@
...
+@RequestMapping(path="/demo")
 public class IndexController {
-    @GetMapping("/")
-    public ModelAndView index() {
-        Map<String, String> model = new HashMap<>();
-        model.put("name", "Alexey");
-        return new ModelAndView("index", model);
+    @Autowired
+    private UserRepository userRepository;
```
Comment (M0rtyMerr) [inline src/main/java/com/epam/controller/IndexController.java:18]: Always prefer all args contructor DI (don't use Autowired)

File: src/main/java/com/epam/controller/IndexController.java [hunk 5]
```diff
@@ -1,19 +1,34 @@
...
+
+    @GetMapping(path="/add")
+    public @ResponseBody String addNewUser (@RequestParam String name) {
```
Comment (M0rtyMerr) [inline src/main/java/com/epam/controller/IndexController.java:21]: Are u sure that GET query should add something? Read about REST queries type
Example of get method:
`
@GetMapping("/count")
public ResponseEntity<...> getQuestionCount() {
    return ResponseEntity.ok(...);
}
`

File: src/main/java/com/epam/controller/IndexController.java [hunk 6]
```diff
@@ -1,19 +1,34 @@
...
+
+
+        User n = new User();
```
Comment (M0rtyMerr) [inline src/main/java/com/epam/controller/IndexController.java:24]: U shouldn't perform such kind of actions in Controllet layer. I told u, that your first step is to implement persistence layer. 
If u want implement controller layer just right now - read about architecture patterns. You should have persistence layer (repositories), Service layer and Controller layer.
The only task of controller - cal method from service, wrap result into ResponseEntity (handle exceptions maybe). This logic can't be placed here.
Btw, do u know about lombok val / var feature? I think it's pretty cool. Google it)

File: .travis.yml (hunk 1 above)
Comment (M0rtyMerr) [reply .travis.yml:5]: https://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

File: src/main/java/com/epam/controller/IndexController.java (hunk 6 above)
Comment (M0rtyMerr) [reply src/main/java/com/epam/controller/IndexController.java:24]: And please, delete useless space lines everywhere

File: src/main/java/com/epam/controller/IndexController.java (hunk 4 above)
Comment (TalkingPet) [reply src/main/java/com/epam/controller/IndexController.java:18]: There is a paragraph called "Constructor-based or setter-based DI?" at the following link to Spring Docs which briefly explains why using constructor is the preferable way to inject your dependencies.
https://docs.spring.io/spring/docs/current/spring-framework-reference/core.html#beans-setter-injection

File: src/main/java/com/epam/controller/IndexController.java [hunk 7]
```diff
@@ -1,19 +1,34 @@
...
+        n.setFirstName(name);
+        userRepository.save(n);
+        return "Saved";
+    }
+
+    @GetMapping(path="/all")
```
Comment (TalkingPet) [inline src/main/java/com/epam/controller/IndexController.java:30]: That's rather ambiguous url for REST... You'll be out of URLs pretty fast if you'll keep following such naming convention (like, what url you'll use for getting all rooms, /all2 ?). In a RESTful application your URL should describe what kind of resource you're accessing - for example if you're trying to get a list of all rooms URL should be '/rooms' and its the same for users, meals etc. You can start with something like [this article](https://restfulapi.net/resource-naming/) to get a better understanding of the subject.

File: src/main/java/com/epam/controller/IndexController.java [hunk 8]
```diff
@@ -1,19 +1,34 @@
...
+    private UserRepository userRepository;
+
+    @GetMapping(path="/add")
+    public @ResponseBody String addNewUser (@RequestParam String name) {
+
+
+        User n = new User();
+        n.setFirstName(name);
+        userRepository.save(n);
+        return "Saved";
```
Comment (TalkingPet) [inline src/main/java/com/epam/controller/IndexController.java:27]: You don't have to return a String to describe status of the request, it's better to use response codes instead. jQuery is able to differentiate success codes (2xx) from errors (4xx, 5xx).

**Pull request closed without merging**